File: 백준/Silver/1748. 수 이어 쓰기 1/수 이어 쓰기 1.py
N = int(input())
arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

ans = 0
for i in arr:
    if N >= 10 ** i:
        ans += i * (10 ** i - 10 ** (i - 1))
    else:
        ans += i * (N - 10 ** (i - 1) + 1)
        break
print(ans)

File: 백준/Silver/1748. 수 이어 쓰기 1/수 이어 쓰기 1.java
import java.util.*;
import java.io.*;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int N = sc.nextInt();

		int tmp = 10, ans = 0;

		while (tmp >= 0) {
			if (N >= Math.pow(10, tmp)-1) {
				ans += (tmp+1) * (int)(Math.min(Math.pow(10, tmp+1)-1, N) - (Math.pow(10, tmp) - 1));
			}
			tmp--;
		}

		System.out.println(ans);

	}
}
